packages/sdk/src/universe/types.ts

```diff
@@ -0,0 +1,74 @@
+export enum AssetClass {
+  ERC20 = "ERC20",
+  ERC721 = "ERC721",
+  ERC1155 = "ERC1155",
+}
+
+export enum OrderSide {
+  BUY,
+  SELL,
+}
+
+export type AssetType = {
+  assetClass: string;
+  data: string;
+};
+
+export type Asset = {
+  assetType: AssetType;
+  value: string;
+};
+
+export type LocalAssetType = {
+  assetClass: string;
+  contract?: string;
+  tokenId?: string;
+};
+
+export type LocalAsset = {
+  assetType: LocalAssetType;
+  value: string;
+};
+
+export type OrderKind = "single-token";
+
+export type Order = {
```

Comment:
The on-chain order is defined as follows:
```
    struct Order {
        address maker;
        LibAsset.Asset makeAsset;
        address taker;
        LibAsset.Asset takeAsset;
        uint salt;
        uint start;
        uint end;
        bytes4 dataType;
        bytes data;
    }
```
What do all the other fields in here represent?

Reply:
```
  kind - Currently we support only "single-token". I've added this property because other exchanges have added it as well.
  type - Arbitrary type that we can use to differentiate order versions in case we need to change the order structure/logic. Currently we use "UNIVERSE_V1"
  side - type of the order:  buy or sell
  signature - the signature of the user. Generated using Order.sign();
  fill - the fill of the order. We support partial filling of ERC1155 buy/sell orders and we use this field to perform ofchain validation whether the order can be filled or not. This behavior mimics the one from the contract. Example: User A lists 5 tokens for 1 ETH. Fill is "0". User B buys 2 of them for 0.4 ETH. Fill gets set to "2". User C buys 1 token for 0.2 ETH. Fill gets set to "3". This way we don't allow user to send invalid purchase orders.
  makeStock - Not used currently, can delete.
  makeBalance - Not used currently, can delete.
```

Reply:
Thanks for the info! The fill field seems to be a stateful parameter (we depend on on-chain fill data to know the fill status of any particular order) - any stateful data belongs in the indexer and not in the SDK.